scripts/test-check-packages.sh

```diff
@@ -98,39 +140,35 @@ for d in test/packages/${PACKAGE_TEST_TYPE:-other}/${PACKAGE_UNDER_TEST:-*}/; do
   package_to_test=$(basename "${d}")
 
   if [ "${PACKAGE_TEST_TYPE:-other}" == "benchmarks" ]; then
-    # It is not used PACKAGE_UNDER_TEST, so all benchmark packages are run in the same loop
-    if [ "${package_to_test}" == "pipeline_benchmark" ]; then
-      rm -rf "${PWD}/build/benchmark-results"
-      elastic-package benchmark pipeline -C "$d" -v --report-format xUnit --report-output file --fail-on-missing
-
-      rm -rf "${PWD}/build/benchmark-results-old"
-      mv "${PWD}/build/benchmark-results" "${PWD}/build/benchmark-results-old"
-
-      elastic-package benchmark pipeline -C "$d" -v --report-format json --report-output file --fail-on-missing
-
-      elastic-package report -C "$d" --fail-on-missing benchmark \
-        --new "${PWD}/build/benchmark-results" \
-        --old "${PWD}/build/benchmark-results-old" \
-        --threshold 1 --report-output-path="${PWD}/build/benchreport"
-    fi
-    if [ "${package_to_test}" == "system_benchmark" ]; then
-      elastic-package benchmark system -C "$d" --benchmark logs-benchmark -v --defer-cleanup 1s
-    fi
-  elif [ "${PACKAGE_TEST_TYPE:-other}" == "with-logstash" ] && [ "${package_to_test}" == "system_benchmark" ]; then
-      elastic-package benchmark system -C "$d" --benchmark logs-benchmark -v --defer-cleanup 1s
-  else
-    if [[ "${SERVERLESS}" == "true" ]]; then
-        # skip system tests
-        elastic-package test asset -C "$d" -v --report-format xUnit --report-output file --defer-cleanup 1s  --test-coverage --coverage-format=generic
-        elastic-package test static -C "$d" -v --report-format xUnit --report-output file --defer-cleanup 1s  --test-coverage --coverage-format=generic
-        # FIXME: adding test-coverage for serverless results in errors like this:
-        # Error: error running package pipeline tests: could not complete test run: error calculating pipeline coverage: error fetching pipeline stats for code coverage calculations: need exactly one ES node in stats response (got 4)
-        elastic-package test pipeline -C "$d" -v --report-format xUnit --report-output file --defer-cleanup 1s
-
-        continue
-    fi
-    # Run all tests
-    # defer-cleanup is set to a short period to verify that the option is available
-    elastic-package test -C "$d" -v --report-format xUnit --report-output file --defer-cleanup 1s --test-coverage --coverage-format=generic
+    # FIXME: There are other packages in test/packages/benchmarks folder that are not tested like rally_benchmark
+    case "${package_to_test}" in
+      pipeline_benchmark|use_pipeline_tests)
+        run_pipeline_benchmark "${package_to_test}" "$d"
+        ;;
```

Comment:
Until now just `pipeline_benchmark` package was tested, other packages under `test/packages/benchmarks/` folder were not tested. One of them is `use_pipeline_tests` package.

The difference that I see between `pipeline_benchmark` and `use_pipeline_tests` is that:
- `pipeline_benchmark` has the benchmark configuration for the data stream at `data_stream/test/_dev/benchmark/pipeline/*`. 
- `use_pipeline_tests` no benchmark configuration.

Tested locally with `use_pipeline_tests` package and it does not fail running the pipeline benchmarks and it creates a benchmark report successfully.

I think it can be added here, WDYT ?